numpy/core/arrayprint.py

```diff
@@ -1218,19 +1228,19 @@ def format_float_positional(x, precision=None, unique=True,
                               sign=sign, pad_left=pad_left,
                               pad_right=pad_right, min_digits=min_digits)
 
-
 class IntegerFormat:
-    def __init__(self, data):
+    def __init__(self, data, sign=None):
         if data.size > 0:
             max_str_len = max(len(str(np.max(data))),
                               len(str(np.min(data))))
+            if sign in ['+', ' ']:
+                max_str_len += 1
```

Comment:
Given an array with all negative numbers, this will cause `sign="-"` to be different from `sign="+"`, which I don't think is wanted.  The additional space is needed for _positive_ values only.